src/casts/specify-representation.rkt

```diff
@@ -60,492 +70,601 @@ but a static single assignment is implicitly maintained.
+          (sr-tagged-array-ref ty TYPE-FN-TAG TYPE-FN-HASH-INDEX)]
+         [else (op$ Print err-msg1) (op$ Exit (Quote 1)) UNDEF-IMDT])]))
+  (begin$
+    (assign$ tag (sr-get-tag ty TYPE-TAG-MASK))
+    (case$ tag
+      [(data:TYPE-GREF-TAG)
+       (assign$ arg-ty
+         (sr-tagged-array-ref ty TYPE-GREF-TAG TYPE-GREF-TYPE-INDEX))
+       (assign$ arg-type-hash (type-hash-access arg-ty))
+       (op$ + (op$ * (Quote 19) arg-type-hash) (Quote 1))]
+      [(data:TYPE-GVECT-TAG)
+       (assign$ arg-ty
+         (sr-tagged-array-ref ty TYPE-GVECT-TAG TYPE-GVECT-TYPE-INDEX))
+       (assign$ arg-type-hash (type-hash-access arg-ty))
+       (op$ + (op$ * (Quote 19) arg-type-hash) (Quote 2))]
+      [(data:TYPE-MREF-TAG)
+       (assign$ arg-ty
+         (sr-tagged-array-ref ty TYPE-MREF-TAG TYPE-MREF-TYPE-INDEX))
+       (assign$ arg-type-hash (type-hash-access arg-ty))
+       (op$ + (op$ * (Quote 19) arg-type-hash) (Quote 3))]
+      [(data:TYPE-MVECT-TAG)
+       (assign$ arg-ty
+         (sr-tagged-array-ref ty TYPE-MVECT-TAG TYPE-MVECT-TYPE-INDEX))
+       (assign$ arg-type-hash (type-hash-access arg-ty))
+       (op$ + (op$ * (Quote 19) arg-type-hash) (Quote 4))]
+      [(data:TYPE-FN-TAG)
+       (assign$ return-ty
+         (sr-tagged-array-ref ty TYPE-FN-TAG TYPE-FN-RETURN-INDEX))
+       (assign$ init-hash-code (type-hash-access return-ty))
+       (assign$ args-count
+         (sr-tagged-array-ref ty TYPE-FN-TAG TYPE-FN-ARITY-INDEX))
+       (assign$ iters-count (op$ + TYPE-FN-FMLS-OFFSET args-count))
+       (op$ +
+         (repeat$
+             (i TYPE-FN-FMLS-OFFSET iters-count)
+             (hash-code init-hash-code)
+           (assign$ arg-type (sr-tagged-array-ref ty TYPE-FN-TAG i))
+           (assign$ arg-type-hash (type-hash-access arg-type))
+           (op$ * (Quote 19) (op$ + hash-code arg-type-hash)))
+         (Quote 5))]
+      [(data:TYPE-TUPLE-TAG)
+       (assign$ elms-count
+         (sr-tagged-array-ref ty TYPE-TUPLE-TAG TYPE-TUPLE-COUNT-INDEX))
+       (assign$ iters-count
+         (op$ + TYPE-TUPLE-ELEMENTS-OFFSET elms-count))
+       (op$ +
+         (repeat$
+             (i TYPE-TUPLE-ELEMENTS-OFFSET iters-count)
+             (hash-code (Quote 0))
+           (assign$ elem-type (sr-tagged-array-ref ty TYPE-TUPLE-TAG i))
+           (assign$ elem-type-hash (type-hash-access elem-type))
+           (op$ * (Quote 19) (op$ + hash-code elem-type-hash)))
+         (Quote 6))]
+      [else (op$ Print err-msg2) (op$ Exit (Quote 1)) UNDEF-IMDT])))
+
+(: get-hashcons-types! (-> (Code-Label Uid)))
+(define (get-hashcons-types!)
+  (: make-code! (-> (Code-Label Uid)))
+  (define (make-code!)
+    (define-track-next-uid!$ hashcons-types)
+    (define hashcons-types-label (Code-Label hashcons-types))
+    (define err-msg (Quote "specify-representation/hashcons-types: switch failure"))
+    (define hashcons-types-c : D0-Code
+      (code$ (ty)
+        (If (op$ <= ty TYPE-MAX-ATOMIC-RT-VALUE)
```

Comment:
You could avoid right creep here with the cond$ macro.

Reply:
It does not help much but done.